src/rpcmasternode.cpp

```diff
@@ -240,16 +240,19 @@ Value masternode(const Array& params, bool fHelp)
 Value listmasternodes(const Array& params, bool fHelp)
 {
     std::string strFilter = "";
+    bool returnNetworkAddress = false;
 
-    if (params.size() == 1) strFilter = params[0].get_str();
+    if (params.size() >= 1) strFilter = params[0].get_str();
+    if (params.size() == 2) returnNetworkAddress = (params[1].get_str() == "true" ? true : false);
 
-    if (fHelp || (params.size() > 1))
+    if (fHelp || (params.size() > 2))
         throw runtime_error(
-            "listmasternodes ( \"filter\" )\n"
+            "listmasternodes ( \"filter\" network )\n"
             "\nGet a ranked list of masternodes\n"
 
             "\nArguments:\n"
             "1. \"filter\"    (string, optional) Filter search text. Partial match by txhash, status, or addr.\n"
+            "2. network     (boolean, optional, default=false) Return the network address of masternodes.\n"
```

Comment:
add \"filter\"   type text there,and should be good

Reply:
I don't understand what you want, can you explain your comment?